specs/ExtendedProcessFailed.md

```diff
@@ -0,0 +1,671 @@
+// Register a handler for the ProcessFailed event.
+// This handler checks the failure kind and tries to:
+//   * Recreate the webview for browser failure and render unresponsive.
+//   * Reload the webview for render failure.
+//   * Reload the webview for frame-only render failure impacting app content.
+//   * Log information about the failure for other failures.
+CHECK_FAILURE(m_webView->add_ProcessFailed(
+    Callback<ICoreWebView2ProcessFailedEventHandler>(
+        [this](ICoreWebView2* sender, ICoreWebView2ProcessFailedEventArgs* argsRaw)
+            -> HRESULT {
+            wil::com_ptr<ICoreWebView2ProcessFailedEventArgs> args = argsRaw;
+            COREWEBVIEW2_PROCESS_FAILED_KIND failureKind;
+            CHECK_FAILURE(args->get_ProcessFailedKind(&failureKind));
+            if (failureKind == COREWEBVIEW2_PROCESS_FAILED_KIND_BROWSER_PROCESS_EXITED)
+            {
+                int button = MessageBox(
+                    m_appWindow->GetMainWindow(),
+                    L"Browser process exited unexpectedly.  Recreate webview?",
+                    L"Browser process exited", MB_YESNO);
+                if (button == IDYES)
+                {
+                    m_appWindow->ReinitializeWebView();
+                }
+            }
+            else if (failureKind == COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_UNRESPONSIVE)
+            {
+                int button = MessageBox(
+                    m_appWindow->GetMainWindow(),
+                    L"Browser render process has stopped responding.  Recreate webview?",
+                    L"Web page unresponsive", MB_YESNO);
+                if (button == IDYES)
+                {
+                    m_appWindow->ReinitializeWebView();
+                }
+            }
+            else if (failureKind == COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_EXITED)
+            {
+                int button = MessageBox(
+                    m_appWindow->GetMainWindow(),
+                    L"Browser render process exited unexpectedly. Reload page?",
+                    L"Web page unresponsive", MB_YESNO);
+                if (button == IDYES)
+                {
+                    CHECK_FAILURE(m_webView->Reload());
+                }
+            }
+
+            // Check the runtime event args implements the newer interface.
+            auto args2 =
+                args.try_query<ICoreWebView2ProcessFailedEventArgs2>();
+            if (!args2)
+            {
+                return S_OK;
+            }
+
+            if (failureKind ==
+                COREWEBVIEW2_PROCESS_FAILED_KIND_FRAME_RENDER_PROCESS_EXITED)
+            {
+                // A frame-only renderer has exited unexpectedly. Check if reload is needed.
+                wil::com_ptr<ICoreWebView2FrameInfoCollection> impactedFrames;
+                wil::com_ptr<ICoreWebView2FrameInfoCollectionIterator> iterator;
+                CHECK_FAILURE(args2->get_ImpactedFramesInfo(&impactedFrames));
```

Comment:
Somehow the word "impacted' doesn't sit well with me. Maybe it's just a dental thing. Would "Affected" be better?

Actually, I think "Lost" is best. Because it's not just that the frames were inconvenienced, they were completely lost. `LostFramesInfo`? Or is this `LostFrameInfos`? (Which is weird, because we didn't lose the infos, we lost the frames.)

Reply:
I like "Lost"